### PR TITLE
Add Modules

### PR DESCRIPTION
**Pull request by muhammadfajri-tif**
### Adding modules that are required for games:

- board: Module for manage and display board
- bot: Module for defining the bot's mechanism
- game: Module for defining core logic of the game such as win/lose/draw condition, etc.
- Menu: Module for managing the flow of all menus that are available and displaying them
- score: Module for managing scoreboard and storing it to storage as a file

### CODE REVIEW

**Comment by muhammadfajri-tif** [comment]
Bantu review, kalau udah oke, approve yak 😸 

@RioAgasta @Peitann

File: lib/board.c [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  for (int i = 0; i < side; i++)
+  {
+    for (int j = 0; j < side; j++)
+    {
+      board[i][j].str = " ";
+    }
+  }
+}
+
+int foo;
```
Comment (RioAgasta) [inline lib/board.c:144]: int foo buat apa?

Comment (muhammadfajri-tif) [reply lib/board.c:144]: OHH HAHAHAHA, sorry sorry, dummy variable, gak kepake lagi kok. Brb, gonna delete that. Tengs for the review!

Comment (muhammadfajri-tif) [reply lib/board.c]: fixed!! @RioAgasta

File: lib/bot.c [hunk 2]
```diff
@@ -0,0 +1,422 @@
+#include <time.h>
+#include "bot.h"
+#include "board.c"
+
+/**
+ * Declare global variable for set winner condition.
+ */
+int winCondition;
```
Comment (RioAgasta) [inline lib/bot.c:8]: var winCondition ini bisa diakses sama modul-modul di file game.c gak?

Comment (muhammadfajri-tif) [reply lib/bot.c:8]: bisa bang, itu di declare diluar function/procedure

here's example, coba manggil `winCondition` di main module
image

here's the output:
image

File: lib/bot.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+int evaluate(Board **tempBoard);
+
+int minimax(Board **tempBoard, int depth, int maxDepth, bool isMax);
+
+Move findBestMove(Board **tempBoard);
+
+/**
+ * Procedure for randomize bot's movement.
+ */
+void randomizeMove(int moves[]);
```
Comment (RioAgasta) [inline lib/bot.h:32]: ralat, module randomizeMove ini gak dipake, boleh hapus aja

**Review by RioAgasta — APPROVED** [review]

File: lib/bot.h (hunk 3 above)
Comment (muhammadfajri-tif) [reply lib/bot.h]: Done. Thanks for the review!

#[2c3933e](https://github.com/muhammadfajri-tif/tubes-03-1b/pull/4/commits/2c3933e47944f085bd29b2d9ae8897ab2fe29e31)

**Comment by muhammadfajri-tif** [comment]
@RioAgasta @Peitann 

di merge ya!! Lemme know if there's something wrong 😺 

Feel free buat open PR lagi

thanks!

File: lib/menu.c
Comment (Peitann) [inline lib/menu.c:1]: paling buat back nya kaya gini (?)
if (*opsiMode != 1 && *opsiMode != 2)
  {
    selectGameMode(1, opsiMode);
  }
  else if (*opsiMode == 0){
  	break;
  }
atau pake switch case
switch (opsiMode){
  	case 1:
  		opsiMode();
  		break;
  	case 2:
  		opsiMode();
  		break;
  	case 3:
  		break;
  	default:
  		printf("Invalid Choice Please Choose a Number Between 1-2 or 0!");
  		break;
  }
  paling gini sih kalo ada yang kurang bilang aja ya sorry kalo ga bantu banyak

**Review by Peitann — COMMENTED** [review]
tolong di check ya yang buat check user input di opsiMode (di comment menu.c)

File: lib/menu.c
Comment (RioAgasta) [reply lib/menu.c:1]: kalo pake switch case kepanjangan. pake if-else if aja
terus jangan pake break, kalo opsi nya 0 panggil modul playMenu lagi aja

Comment (Peitann) [reply lib/menu.c:1]: okeh yo thanks

**Pull request merged**